### PR TITLE
ACS-276 tas test for rendition version

### PR DESCRIPTION
**Pull request by nithinnambiar**
Added TAS test for rendition version REST API. The tests are executed as part of "REST API TAS tests part3"  tas tests. See tests/tas-restapi/src/test/resources/test-suites/part3-suite.xml.

https://alfresco.atlassian.net/browse/ACS-276

### CODE REVIEW

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/renditions/RenditionVersionTests.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
+package org.alfresco.rest.renditions;
```
Comment (montgolfiere) [inline packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/renditions/RenditionVersionTests.java:1]: Add license header

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/renditions/RenditionVersionTests.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        // version update
+        restClient.authenticateUser(user).withCoreAPI().usingNode(file).updateNodeContent(sampleFile);
+        restClient.assertStatusCodeIs(HttpStatus.OK);
+
+        STEP("2. Create the pdf rendition of txt file using RESTAPI");
+        restClient.withCoreAPI().usingNode(file).createNodeVersionRendition("pdf", "1.1");
+        restClient.assertStatusCodeIs(HttpStatus.ACCEPTED);
+
+        STEP("3. Verify pdf rendition of txt file is created");
+        restClient.withCoreAPI().usingNode(file).getNodeVersionRenditionUntilIsCreated("pdf", "1.1").assertThat()
```
Comment (montgolfiere) [inline packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/renditions/RenditionVersionTests.java:65]: Requesting a rendition is async. What happens if it is still in progress (ie. not created yet) ?

**Review by montgolfiere — COMMENTED** [review]
@nithinnambiar I've add a few initial comments.

Which suite runs this test in Travis ?


Also, do we have any existing tests for live renditions (vs version renditions) ? If so, shouldn't we combine into one class since the actual test behaviour will be very similar (albeit slightly different url endoints).

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/renditions/RenditionVersionTests.java (hunk 1 above)
Comment (nithinnambiar) [reply packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/renditions/RenditionVersionTests.java:1]: done.

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/renditions/RenditionVersionTests.java (hunk 2 above)
Comment (nithinnambiar) [reply packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/renditions/RenditionVersionTests.java:65]: Step 3 retries few times if it is still not created.

**Comment by nithinnambiar** [comment]
> @nithinnambiar Thanks for taking on. I've add a few initial comments. Who else in Team 3 will review ?
> 
> For future reference, can you please add a link to the PR (&/or JIRA) description with link to the test suite (and Travis job) that actually runs this specific test class.
> 
> Also, do we have any existing tests for renditions on the live node (vs version renditions) ?
> 
> If so, shouldn't we combine into one class since the actual logic (test behaviour) will be almost identical, albeit slightly different url endpoints.

**Comment by nithinnambiar** [comment]
> @nithinnambiar Thanks for taking on. I've add a few initial comments. Who else in Team 3 will review ?
> 
> For future reference, can you please add a link to the PR (&/or JIRA) description with link to the test suite (and Travis job) that actually runs this specific test class.
> 
> Also, do we have any existing tests for renditions on the live node (vs version renditions) ?
> 
> If so, shouldn't we combine into one class since the actual logic (test behaviour) will be almost identical, albeit slightly different url endpoints.
I have updated the description.
The test for rendition live nodes are scattered across multiple classes and some of the tests are not strictly aligned to the end points. I think from the maintainability perspective it make sense to keep them separate.

**Review by amohammedalfresco — APPROVED** [review]
LGTM!

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/renditions/RenditionVersionTests.java (hunk 2 above)
Comment (montgolfiere) [reply packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/renditions/RenditionVersionTests.java:80]: 👍

**Pull request merged**